core/src/main/java/de/jplag/normalization/Edge.java

```diff
@@ -0,0 +1,33 @@
+package de.jplag.normalization;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import de.jplag.semantics.Variable;
+
+class Edge {
+    private Set<EdgeItem> items;
+    private boolean isVariableFlow;
+    private boolean isVariableReverseFlow;
+
+    Edge() {
+        items = new HashSet<>();
+        isVariableFlow = false;
+    }
+
+    boolean isVariableFlow() {
+        return isVariableFlow;
```

Comment:
You could consider just storing `EdgeType type` as an instance variable in `addItem` (instead of separate variables for `isVariableFlow` and `isVariableReverseFlow`), then change this to
```suggestion
        return type == EdgeType.VARIABLE_FLOW
```

Reply:
Edge is actually a multiple edge, so it can have multiple types, so this doesn't really work. Bad naming on my part, I've renamed `Edge` to `MultipleEdge` and `EdgeItem` to `Edge` to better reflect what's going on